modules/formulize/include/functions.php

```diff
@@ -8051,3 +8073,50 @@ function generateSelfReferenceExclusionSQL($entry_id, $fid, $sourceFid, $ele_val
 	}
 	return $selfReferenceExclusion;
 }
+
+/**
+ * Record information in a log file. One log file per day. Log file location is a Formulize preference. Log file storage duration is a Formulize preference.
+ *
+ * @param array $data Key-Value pairs that should be written to the log entry
+ * @return int|boolean Returns the number of bytes written to the file, or false on failure
+ */
+function writeToFormulizeLog($data) {
+
+	// initialize the configuration settings
+	static $formulizeConfig = false;
+	static $formulizeLogFileLocation = XOOPS_ROOT_PATH.'/logs';
+	static $formulizeLogFileStorageDuration = 168;
+	static $logFilesCleanedUp = false;
+	if(!$formulizeConfig) {
+		$config_handler = xoops_gethandler('config');
+		$formulizeConfig = $config_handler->getConfigsByCat(0, getFormulizeModId());
+		$formulizeLogFileLocation = $formulizeConfig['formulizeLogFileLocation'];
+		$formulizeLogFileStorageDuration = $formulizeConfig['formulizeLogFileStorageDuration'];
+	}
+
+	// check if log file location is valid
+	if(!$formulizeLogFileLocation OR !is_dir($formulizeLogFileLocation)) { return false; }
+
+	// cleanup old log files (last param requires seconds) -- only once per page load
+	if(!$logFilesCleanedUp) {
+		formulize_scandirAndClean($formulizeLogFileLocation, 'formulize_log_', $formulizeLogFileStorageDuration * 60 * 60);
+		$logFilesCleanedUp = true;
+	}
+
+	// prepend certain metadata to the data we're logging
+	// UNIQUE ID is only available if the server has mod_unique_id turned on
+	$data = array(
+		'microtime' => microtime(true),
+		'request_id' => (isset($_SERVER['UNIQUE_ID']) ? $_SERVER['UNIQUE_ID'] : 'mod_unique_id is not enabled'),
```

Comment:
The fallback case here won't fly with the log processing. I think we should fallback to the microtime in this case.

Reply:
Upon further reflection we could fallback to https://www.php.net/manual/en/function.uniqid.php

Reply:
We will generate a PHP unique id as a static variable and reuse that, if there is no value for $_SERVER['unique_id']